tests/BaseTest.php

```diff
@@ -0,0 +1,130 @@
+        $this->assertCount(4, $options);
+        $this->assertSame(5, $options['test-option']);
+        $this->assertSame(false, $options[Base::OPT_SLAVES_SYNC_ENABLED]);
+        $this->assertSame(10, $options[Base::OPT_SLAVES_SYNC_REQUIRED_COUNT]);
+        $this->assertSame(100, $options[Base::OPT_SLAVES_SYNC_TIMEOUT]);
+    }
+
+    public function testGetRedisClient()
+    {
+        $queue = new TestQueueImplementation($redis = $this->getRedisMock(), $name = 'test');
+
+        $this->assertSame($redis, $queue->getRedisClient());
+    }
+
+    public function testGetName()
+    {
+        $queue = new TestQueueImplementation($this->getRedisMock(), $name = 'test');
+
+        $this->assertSame($name, $queue->getName());
+    }
+
+    public function testWaitForSlaveSyncDisabled()
+    {
+        $redis = $this->getRedisMock();
+        $redis->shouldReceive('wait')->never();
+        $queue = new TestQueueImplementation($redis, $name = 'test', [Base::OPT_SLAVES_SYNC_REQUIRED_COUNT => 5]);
+
+        $queue->testWaitForSlaveSync();
+        Mockery::close();
+    }
+
+    public function testWaitForSlaveSyncExact()
+    {
+        $redis = $this->getRedisMock();
+        $redis->shouldReceive('wait')->once()->with($required = 3, 100)->andReturn($synced = 5);
+        $queue = new TestQueueImplementation($redis, $name = 'test', [
+            Base::OPT_SLAVES_SYNC_ENABLED        => true,
+            Base::OPT_SLAVES_SYNC_REQUIRED_COUNT => $required,
+        ]);
+
+        $queue->testWaitForSlaveSync();
+        Mockery::close();
+    }
+
+    public function testWaitForSlaveSyncEnough()
+    {
+        $redis = $this->getRedisMock();
+        $redis->shouldReceive('wait')->once()->with($required = 3, 100)->andReturn($synced = 5);
+        $queue = new TestQueueImplementation($redis, $name = 'test', [
+            Base::OPT_SLAVES_SYNC_ENABLED        => true,
+            Base::OPT_SLAVES_SYNC_REQUIRED_COUNT => $required,
+        ]);
+
+        $queue->testWaitForSlaveSync();
+        Mockery::close();
+    }
+
+    public function testWaitForSlaveSyncNotEnough()
+    {
+        $redis = $this->getRedisMock();
+        $redis->shouldReceive('wait')->once()->with($required = 3, 100)->andReturn($synced = 2);
+        $queue = new TestQueueImplementation($redis, $name = 'test', [
+            Base::OPT_SLAVES_SYNC_ENABLED        => true,
+            Base::OPT_SLAVES_SYNC_REQUIRED_COUNT => $required,
+        ]);
+
+        try {
+            $queue->testWaitForSlaveSync();
+        } catch (Exception\NotEnoughSlavesSynced $e) {
+            if ($e->getMessage() !== sprintf('Required: %d, synced: %d', $required, $synced)) {
+                throw $e;
+            }
+        }
+
+        Mockery::close();
+    }
+
+    /**
+     * @return Mockery\Mock|ClientInterface
+     */
+    private function getRedisMock()
+    {
+        return Mockery::mock('PhpRQ\ClientInterface');
+    }
+
+}
+
+class TestQueueImplementation extends Base
```

Comment:
já tady ale netestuju private metody, ne? Jenom tu protected waitForSlaveSync ... a jako jo, šlo by to udělat helperem, ale přijde mi to jako zbytečný - to by jsi ho pak musel všude předávat jako dependency (což by zesložitilo instancování) a reálně by fakt měl jen jednu simple metodu ... myslíš, že to za to stojí?